File: monsoon3.0/src/main/java/wtf/monsoon/api/command/Command.java
package wtf.monsoon.api.command;

import net.minecraft.client.Minecraft;
import wtf.monsoon.Wrapper;

/**
 * @author Surge
 * @since 28/07/2022
 */
public abstract class Command {

    // Name of the command
    private final String name;

    // Minecraft instance
    public Minecraft mc = Wrapper.getMinecraft();

    public Command(String name) {
        this.name = name;
    }

    /**
     * Executes the command.
     *
     * @param args The arguments we have supplied.
     */
    public abstract void execute(String[] args);

    /**
     * Gets the name of the command.
     *
     * @return The name of the command.
     */
    public String getName() {
        return name;
    }

}

File: 3.x-server/src/main/kotlin/wtf/monsoon/server/discord/command/Command.kt
package wtf.monsoon.server.discord.command

import org.javacord.api.interaction.SlashCommand
import org.javacord.api.interaction.SlashCommandBuilder
import org.javacord.api.interaction.SlashCommandInteraction

abstract class Command(var name: String, var description: String) {
    open val builder: SlashCommandBuilder
        get() = SlashCommand.with(name, description)

    abstract fun execute(interaction: SlashCommandInteraction)
}